iceoryx2/tests/service_event_tests.rs

```diff
@@ -365,6 +370,182 @@ mod service_event {
+                    }
+                }));
+            }
+
+            for _ in 0..NUMBER_OF_NOTIFIER_THREADS {
+                s.spawn(|| {
+                    barrier.wait();
+
+                    while keep_running.load(Ordering::Relaxed) {
+                        let notifier = sut.notifier().create().unwrap();
+                        assert_that!(notifier.notify_with_custom_event_id(EVENT_ID), is_ok);
+                    }
+                });
+            }
+
+            for thread in listener_threads {
+                thread.join().unwrap();
+            }
+
+            keep_running.store(false, Ordering::Relaxed);
+        });
+    }
+
+    #[test]
+    // TODO iox2-139
+    #[ignore]
+    fn concurrent_reconnecting_listener_can_wait_for_triggering_notifiers<Sut: Service>() {
+        let _watch_dog = Watchdog::new(Duration::from_secs(1));
+
+        const NUMBER_OF_LISTENER_THREADS: usize = 2;
+        const NUMBER_OF_NOTIFIER_THREADS: usize = 2;
+        const NUMBER_OF_ITERATIONS: usize = 50;
+        const EVENT_ID: EventId = EventId::new(558);
+
+        let keep_running = AtomicBool::new(true);
+        let service_name = generate_name();
+        let barrier = Barrier::new(NUMBER_OF_LISTENER_THREADS + NUMBER_OF_NOTIFIER_THREADS);
+
+        let sut = Sut::new(&service_name)
+            .event()
+            .max_listeners(NUMBER_OF_LISTENER_THREADS * 2)
+            .max_notifiers(NUMBER_OF_NOTIFIER_THREADS)
+            .create()
+            .unwrap();
+
+        std::thread::scope(|s| {
+            let mut listener_threads = vec![];
+            for _ in 0..NUMBER_OF_LISTENER_THREADS {
+                listener_threads.push(s.spawn(|| {
+                    barrier.wait();
+
+                    let mut counter = 0;
+                    let mut listener = sut.listener().create().unwrap();
+                    while counter < NUMBER_OF_ITERATIONS {
+                        let event_ids = listener.blocking_wait().unwrap();
+                        if !event_ids.is_empty() {
+                            counter += 1;
+                            for id in event_ids {
+                                assert_that!(*id, eq EVENT_ID);
+                            }
+                            listener = sut.listener().create().unwrap();
+                        }
+                    }
+                }));
+            }
+
+            for _ in 0..NUMBER_OF_NOTIFIER_THREADS {
+                s.spawn(|| {
+                    let notifier = sut.notifier().create().unwrap();
+                    barrier.wait();
+
+                    while keep_running.load(Ordering::Relaxed) {
+                        assert_that!(notifier.notify_with_custom_event_id(EVENT_ID), is_ok);
+                    }
+                });
+            }
+
+            for thread in listener_threads {
+                thread.join().unwrap();
+            }
+
+            keep_running.store(false, Ordering::Relaxed);
+        });
+    }
+
```

Comment:
How about a test with concurrently reconnecting listeners waiting for concurrently reconnection notifiers?

Reply:
I added this as a todo for #139 . This test would now completely fail in multiple places and those test itself can also be bug-prone, therefore I would like to write it once it should theoretically work. Otherwise, I have a test that is bug prone and tests something bug prone and fails for random reasons.